src/zkscript/fields/fq12_2_over_3_over_2.py

```diff
@@ -376,18 +416,30 @@ def frobenius_odd(
         clean_constant: bool | None = None,
         is_constant_reused: bool | None = None,
     ) -> Script:
-        r"""Frobenius for odd powers n = 2k + 1 in F_q^12.
-
-        Then, Frobenius is computed via the isomorphism: F_q^12 \sim F_q^2[u,v] / (u^2 - v, v^3 - NON_RESIDUE_OVER_FQ2)
-        \sim F_q^2[t] / (t^6 - NON_RESIDUE_OVER_FQ2)
-
-        Input parameters:
-            - Stack: q .. X
-            - Altstack:
-        Output:
-            - X**q**(n)
-        Assumption on data:
-            - X is passed as as a sixtuple of elements of Fq2. Namely, X = a b c d e f
+        """Frobenius for odd powers n = 2k + 1 in F_q^12.
+
+        Frobenius is computed via the isomorphism:
+        F_q^12 ~ F_q^2[u,v] / (u^2 - v, v^3 - non_residue_over_fq2)
+            ~ F_q^2[t] / (t^6 - non_residue_over_fq2)
+
+        Stack input:
+            - stack:    [q, ..., x := (x0, x1, ..., x11)], `x` is a sixtuple of elements of F_q^2
+            - altstack: []
+
+        Stack output:
+            - stack:    [q, ..., x^q^n]
+            - altstack: []
+
+        Args:
+            n (int): Frobenius odd power.
+            take_modulo (bool): If `True`, the result is reduced modulo `q`.
+            check_constant (bool | None): If `True`, check if `q` is valid before proceeding. Defaults to `None`.
+            clean_constant (bool | None): If `True`, remove `q` from the bottom of the stack. Defaults to `None`.
+            is_constant_reused (bool | None, optional): If `True`, `q` remains as the second-to-top element on the stack
+                after execution. Defaults to `None`.
+
+        Returns:
+            Script to compute the Frobenius endomorphism for odd powers of an element in F_q^12.
         """
         assert n % 2 == 1
         assert n % 12 != 0
```

Comment:
do we need this assert? seems useless (n is odd, so it is not divisible by 12)

Reply:
Removed